### PR TITLE
[IMP] extract_api: Add bank statement documentation

### PR DESCRIPTION
**Pull request by jmak-odoo**
- Updated documentation to include details on the new bank statement extraction via OCR.
- Removed mention of unsupported discount, product, and unit features from invoice_lines feature.

Task-3895452

### CODE REVIEW

**Review by AntoineVDV — APPROVED** [review]
@robodoo delegate+

File: content/developer/reference/extract_api.rst [hunk 1]
```diff
@@ -465,19 +468,47 @@ represents an invoice line. Each value follows a
...
+It is returned as a list of dictionaries where each dictionary represents a bank statement line.
+Each value follows a :ref:`latestextract_api/get_result/feature_result` structure.
+
+.. code-block:: js
+
+    "bank_statement_lines": [
+        {
+            "amount": feature_result,
+            "description": feature_result,
+            "date": feature_result,
```
Comment (lobaudoux) [inline content/developer/reference/extract_api.rst:508]: It's actually not a `feature_result` for bank statements, we return the value directly.

**Review by lobaudoux — APPROVED** [review]
LGTM, thanks

**Comment by jmak-odoo** [comment]
@robodoo r+

**Comment by jmak-odoo** [comment]
@robodoo retry

**Comment by jmak-odoo** [comment]
@robodoo r+

**Pull request closed without merging**